web/src/engine/interfaces/src/prediction/languageProcessor.interface.ts

```diff
@@ -0,0 +1,75 @@
+///<reference types="@keymanapp/models-types" />
+
+import { EventEmitter } from "eventemitter3";
+import { OutputTarget } from "@keymanapp/keyboard-processor";
+
+export class ReadySuggestions {
+  suggestions: Suggestion[];
+  transcriptionID: number;
+
+  constructor(suggestions: Suggestion[], id: number) {
+    this.suggestions = suggestions;
+    this.transcriptionID = id;
+  }
+}
+
+/**
+ * Corresponds to the 'suggestionsready' LanguageProcessor event.
+ */
+export type ReadySuggestionsHandler = (prediction: ReadySuggestions) => boolean;
+
+export type InvalidateSourceEnum = 'new' | 'context';
+
+/**
+ * Corresponds to the 'invalidatesuggestions' LanguageProcessor event.
+ */
+export type InvalidateSuggestionsHandler = (source: InvalidateSourceEnum) => boolean;
+
+export type StateChangeEnum = 'active' | 'configured' | 'inactive';
+/**
+ * Corresponds to the 'statechange' LanguageProcessor event.
+ */
+export type StateChangeHandler = (state: StateChangeEnum) => any;
+
+/**
+ * Covers 'tryaccept' events.
+ */
+export type TryUIHandler = (source: string, returnObj: { shouldSwallow: boolean }) => boolean;
+
+export interface LanguageProcessorEventMap {
```

Comment:
Great way to move forward on this and preserve modularization for the different moving parts.  ❤️